test/e2e/framework/framework.go

```diff
@@ -0,0 +1,319 @@
+	}
+
+	Logf("namespace %v deletion completed in %s", namespace, time.Now().Sub(startTime))
+	return nil
+}
+
+// InstallProvisioner installs the Volume Provisioner as a deployment into the given namespace
+func (f *Framework) InstallProvisioner(namespace string) error {
+	return nil
+
+}
+
+// BeforeEach gets a client and makes a namespace.
+func (f *Framework) BeforeEach() {
+	// The fact that we need this feels like a bug in ginkgo.
+	// https://github.com/onsi/ginkgo/issues/222
+	f.cleanupHandle = AddCleanupAction(f.AfterEach)
+
+	if f.ClientSet == nil {
+		By("Creating a kubernetes client")
+		config, err := clientcmd.BuildConfigFromFlags("", TestContext.KubeConfig)
+		Expect(err).NotTo(HaveOccurred())
+		f.ClientSet, err = clientset.NewForConfig(config)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	if TestContext.Namespace == "" {
+		By("Building a namespace api object")
+		namespace, err := f.CreateNamespace(f.BaseName, map[string]string{
+			"e2e-framework": f.BaseName,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		f.Namespace = namespace
+	} else {
+		By(fmt.Sprintf("Getting existing namespace %q", TestContext.Namespace))
+		namespace, err := f.ClientSet.CoreV1().Namespaces().Get(TestContext.Namespace, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		f.Namespace = namespace
+	}
+
+	if f.IsBackup {
+		f.BlockStorageClient = f.createStorageClient()
+	}
+	if !f.ProvisionerInstalled {
+		err := f.InstallProvisioner(f.Namespace.Name)
+		Expect(err).NotTo(HaveOccurred())
+		f.ProvisionerInstalled = true
+	}
+}
+
+// AfterEach deletes the namespace(s).
+func (f *Framework) AfterEach() {
+	RemoveCleanupAction(f.cleanupHandle)
+
+	nsDeletionErrors := map[string]error{}
+
+	// Whether to delete namespace is determined by 3 factors: delete-namespace flag, delete-namespace-on-failure flag and the test result
+	// if delete-namespace set to false, namespace will always be preserved.
+	// if delete-namespace is true and delete-namespace-on-failure is false, namespace will be preserved if test failed.
+	if TestContext.DeleteNamespace && (TestContext.DeleteNamespaceOnFailure || !CurrentGinkgoTestDescription().Failed) {
+		for _, ns := range f.namespacesToDelete {
+			By(fmt.Sprintf("Destroying namespace %q for this suite.", ns.Name))
+			if err := f.DeleteNamespace(ns.Name, 5*time.Minute); err != nil {
+				nsDeletionErrors[ns.Name] = err
+			}
+		}
+	}
+
+	// if we had errors deleting, report them now.
+	if len(nsDeletionErrors) != 0 {
+		messages := []string{}
+		for namespaceKey, namespaceErr := range nsDeletionErrors {
+			messages = append(messages, fmt.Sprintf("Couldn't delete ns: %q: %s (%#v)", namespaceKey, namespaceErr, namespaceErr))
+		}
+		Failf(strings.Join(messages, ","))
+	}
+	f.ProvisionerInstalled = false
+}
+
+func (f *Framework) createStorageClient() coreOCI.BlockstorageClient {
+	By("Creating an OCI block storage client")
+	configPath, ok := os.LookupEnv("OCICONFIG_VAR")
```

Comment:
Declare as global variable and use it throughout the code (if used anywhere else). It is good to have a set of global variables that defines all of the environment variables being set and required for the test